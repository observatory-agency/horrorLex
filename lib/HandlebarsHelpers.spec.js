const HandlebarsHelpers = require('./HandlebarsHelpers');

describe('HandlebarsHelpers', () => {
  let hbsMock;
  let optionsMock;
  beforeEach(() => {
    hbsMock = jest.fn();
    optionsMock = {
      fn: jest.fn(({ page }) => page),
      inverse: jest.fn((a) => a),
      data: {
        root: {
          page: {
            curr: 5,
          },
        },
      },
    };
  });

  describe('constructor', () => {
    it('should return an instance of HandlebarsHelpers', () => {
      expect(new HandlebarsHelpers(hbsMock)).toBeInstanceOf(HandlebarsHelpers);
    });
  });

  describe('envHelper', () => {
    it('should call options.fn', () => {
      new HandlebarsHelpers(hbsMock).envHelper('development', optionsMock);
      expect(optionsMock.fn).toHaveBeenCalled();
    });
    it('should call options.inverse', () => {
      new HandlebarsHelpers(hbsMock).envHelper('production', optionsMock);
      expect(optionsMock.inverse).toHaveBeenCalled();
    });
  });

  describe('pageActiveHelper', () => {
    it('should call options.fn', () => {
      new HandlebarsHelpers(hbsMock).pageActiveHelper(5, optionsMock);
      expect(optionsMock.fn).toHaveBeenCalled();
    });
    it('should call options.inverse', () => {
      new HandlebarsHelpers(hbsMock).pageActiveHelper(6, optionsMock);
      expect(optionsMock.inverse).toHaveBeenCalled();
    });
  });

  describe('pageNumberHelper', () => {
    describe('when handling many pages', () => {
      it('should a string with appropriate page numbers and skip characters', () => {
        const pages = Array.from({ length: 20 }, (a, i) => i + 1);
        const helper = new HandlebarsHelpers(hbsMock).pageNumberHelper(pages, optionsMock);
        expect(helper).toMatchSnapshot();
      });
    });
    describe('when the current page is the first page', () => {
      beforeEach(() => {
        optionsMock.data.root.page.curr = 1;
      });
      it('should a string with appropriate page numbers and skip characters', () => {
        const pages = Array.from({ length: 20 }, (a, i) => i + 1);
        const helper = new HandlebarsHelpers(hbsMock).pageNumberHelper(pages, optionsMock);
        expect(helper).toMatchSnapshot();
      });
    });
    describe('when the current page is the last page', () => {
      beforeEach(() => {
        optionsMock.data.root.page.curr = 20;
      });
      it('should a string with appropriate page numbers and skip characters', () => {
        const pages = Array.from({ length: 20 }, (a, i) => i + 1);
        const helper = new HandlebarsHelpers(hbsMock).pageNumberHelper(pages, optionsMock);
        expect(helper).toMatchSnapshot();
      });
    });
    describe('when handling 3 pages', () => {
      it('should a string with appropriate page numbers and skip characters', () => {
        const pages = Array.from({ length: 3 }, (a, i) => i + 1);
        const helper = new HandlebarsHelpers(hbsMock).pageNumberHelper(pages, optionsMock);
        expect(helper).toMatchSnapshot();
      });
    });
    describe('when handling 1 page', () => {
      it('should a string with appropriate page numbers and skip characters', () => {
        const pages = Array.from({ length: 1 }, (a, i) => i + 1);
        const helper = new HandlebarsHelpers(hbsMock).pageNumberHelper(pages, optionsMock);
        expect(helper).toMatchSnapshot();
      });
    });
  });

  describe('pageSkipHelper', () => {
    it('should call options.fn', () => {
      new HandlebarsHelpers(hbsMock).pageSkipHelper('...', optionsMock);
      expect(optionsMock.fn).toHaveBeenCalled();
    });
    it('should call options.inverse', () => {
      new HandlebarsHelpers(hbsMock).pageSkipHelper(1, optionsMock);
      expect(optionsMock.inverse).toHaveBeenCalled();
    });
  });
});
