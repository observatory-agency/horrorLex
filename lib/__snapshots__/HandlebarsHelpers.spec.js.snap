// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HandlebarsHelpers pageNumberHelper when handling 1 page should a string with appropriate page numbers and skip characters 1`] = `"1"`;

exports[`HandlebarsHelpers pageNumberHelper when handling 3 pages should a string with appropriate page numbers and skip characters 1`] = `"123"`;

exports[`HandlebarsHelpers pageNumberHelper when handling many pages should a string with appropriate page numbers and skip characters 1`] = `"1...34567...20"`;

exports[`HandlebarsHelpers pageNumberHelper when the current page is the first page should a string with appropriate page numbers and skip characters 1`] = `"12345...20"`;

exports[`HandlebarsHelpers pageNumberHelper when the current page is the last page should a string with appropriate page numbers and skip characters 1`] = `"1...1617181920"`;
